### PR TITLE
Dev team1 sharebutton 

### PR DESCRIPTION
**Pull request by samaritan-mt**
Share button with automatic copy to the clipboard : 
Added a single post page based on the project component and whose route is : post with id query parameter for post fetching 
changed the data structure of the post interface to include a postID (unique id for the post which is ref for firebase) 
The send button now copies the single page of the post to the clipboard

### CODE REVIEW

**Comment by samaritan-mt** [comment]
Can you please check  the functions of this PR before approving @Hendercraft @BenjaminCDP @Nellou90 for anything I could have missed and just one thing the posts that have been posted before the adding of postID on the Post interface won't be accessible through the send button (hence my message about flushing the db) but if you have their refs you can see them individually using the route : post?id=ref. 
Normally this won't happen with future posts as the ref is set as post ID automatically on the post service. 
Thank you

**Review by Nellou90 — APPROVED** [review]

File: AgileFront/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -22,11 +24,16 @@ const routes: Routes = [
     component: CreatePostComponent,
     canActivate: [authGuard],
   },
+  {
+    path: 'post',
+    component: ProjectComponent,
+    canActivate: [authGuard],
```
Comment (Hendercraft) [inline AgileFront/src/app/app-routing.module.ts:30]: We'll might want to remove tha guard, but add protection to redirect the user to the login page if he try to like or comment a post

File: AgileFront/src/app/services/post.ts [hunk 2]
```diff
@@ -3,4 +3,5 @@ export interface Post {
   body?: string;
   imageURL: string;
   userId: string;
+  postId?: string;
```
Comment (Hendercraft) [inline AgileFront/src/app/services/post.ts:6]: I get why you do that, might be a more elegent way to do so, but whatev, one week left anyway

**Review by Hendercraft — APPROVED** [review]
Seems good, just check the comment i made

File: AgileFront/src/app/app-routing.module.ts (hunk 1 above)
Comment (samaritan-mt) [reply AgileFront/src/app/app-routing.module.ts:30]: I'll implement it with the like button no worries

File: AgileFront/src/app/services/post.ts (hunk 2 above)
Comment (samaritan-mt) [reply AgileFront/src/app/services/post.ts:6]: Yeah I said it in the comment on the PR

**Pull request merged**